constants/ecp5_platforms.py

```
@@ -123,9 +123,12 @@ class ECP5BG756Platform(LatticeECP5Platform):
         default_clk = "clk"
         default_rst = "rst"
 
+        clk_pin = pins.named_pin(ecp5_bg756_pclk)
+        if clk_pin is None:
```

this is required only to satisfy the typechecker, right?

---

I would much rather throw an exception in the `PinManager`, and keep this code clean.